In [1]:
import numpy as np

------------
## Loading data 

In [2]:
data = np.genfromtxt('fit.csv', delimiter=',', dtype=None, encoding=None)
print(data)

[('06-10-2017', 5464, 'Neutral', 181, 5, 'Inactive')
 ('07-10-2017', 6041, 'Sad', 197, 8, 'Inactive')
 ('08-10-2017',   25, 'Sad',   0, 5, 'Inactive')
 ('09-10-2017', 5461, 'Sad', 174, 4, 'Inactive')
 ('10-10-2017', 6915, 'Neutral', 223, 5, 'Active')
 ('11-10-2017', 4545, 'Sad', 149, 6, 'Inactive')
 ('12-10-2017', 4340, 'Sad', 140, 6, 'Inactive')
 ('13-10-2017', 1230, 'Sad',  38, 7, 'Inactive')
 ('14-10-2017',   61, 'Sad',   1, 5, 'Inactive')
 ('15-10-2017', 1258, 'Sad',  40, 6, 'Inactive')
 ('16-10-2017', 3148, 'Sad', 101, 8, 'Inactive')
 ('17-10-2017', 4687, 'Sad', 152, 5, 'Inactive')
 ('18-10-2017', 4732, 'Happy', 150, 6, 'Active')
 ('19-10-2017', 3519, 'Sad', 113, 7, 'Inactive')
 ('20-10-2017', 1580, 'Sad',  49, 5, 'Inactive')
 ('21-10-2017', 2822, 'Sad',  86, 6, 'Inactive')
 ('22-10-2017',  181, 'Sad',   6, 8, 'Inactive')
 ('23-10-2017', 3158, 'Neutral',  99, 5, 'Inactive')
 ('24-10-2017', 4383, 'Neutral', 143, 4, 'Inactive')
 ('25-10-2017', 3881, 'Neutral', 125, 5, 'Inactive')
 (

In [3]:
date=data['f0']
step_count=data['f1']
mood=data['f2']
calories_burned=data['f3']
hours_of_sleep=data['f4']
active=data['f5']

-------------------

## Visualizing Data

#### Seperating coloumns

In [4]:
mood

array(['Neutral', 'Sad', 'Sad', 'Sad', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Sad', 'Sad', 'Sad', 'Sad', 'Happy', 'Sad', 'Sad', 'Sad', 'Sad',
       'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral',
       'Happy', 'Neutral', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Neutral', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Neutral',
       'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Neutral', 'Sad', 'Happy', 'Happy', 'Happy',
       'Happy', 'Happy', 'Happy', 'Happy', 'Sad', 'Neutral', 'Neutral',
       'Sad', 'Sad', 'Neutral', 'Neutral', 'Happy', 'Neutral', 'Neutral',
       'Sad', 'Neutral', 'Sad', 'Neutral', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Sad', 'Happy', 'Neutral', 'Happy', 'Neutral', 'Sad', 'Sad', 'Sad',
       'Neutral', 'Neutral', 'Sad', 'Sad', 'Happy', 'Neutral', 'Neutral',
       'Happy'], dtype='<U7')

In [5]:
calories_burned

array([181, 197,   0, 174, 223, 149, 140,  38,   1,  40, 101, 152, 150,
       113,  49,  86,   6,  99, 143, 125, 129,   6,   9,  10,  72, 150,
       141, 156,  57,  72,  17, 181, 197, 131, 154, 137, 193,  19, 101,
       139, 164, 137,  22,  17,   9, 145, 192, 146, 234, 167,  16,  17,
        32,  35, 220, 116,  23,  44, 131,  86, 194,  60, 121,  76,  93,
        53,  25, 227, 125, 243,  14,  39,  55, 158,   7, 213, 116, 129,
        21,  28,  16, 180, 138, 176,  99,   5,   1,  10,  47,  84,  23,
         4,   0,   0,   0,   0])

In [6]:
hours_of_sleep

array([5, 8, 5, 4, 5, 6, 6, 7, 5, 6, 8, 5, 6, 7, 5, 6, 8, 5, 4, 5, 6, 8,
       5, 6, 5, 8, 5, 4, 5, 4, 5, 4, 3, 2, 9, 5, 6, 4, 5, 8, 4, 5, 6, 5,
       6, 5, 6, 5, 6, 5, 6, 7, 6, 7, 6, 5, 6, 7, 8, 8, 7, 8, 5, 4, 3, 3,
       4, 5, 5, 5, 3, 4, 4, 5, 5, 5, 5, 5, 5, 4, 3, 4, 5, 5, 4, 5, 3, 3,
       3, 2, 3, 2, 8, 5, 5, 5])

In [7]:
active

array(['Inactive', 'Inactive', 'Inactive', 'Inactive', 'Active',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Active', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Active', 'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Active',
       'Inactive', 'Inactive', 'Inactive', 'Inactive', 'Inactive',
       'Active', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Inactive', 'Inactive', 'Inactive',
       'Inactive', 'Inactive', 'Inactive', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Active', 'Active', 'Active', 'Inactive', 'Active', 'Active',
       'Inactive', 'Active', 'Active', 'Active', 'Active', 'Active',
       'Inactive', 'Active', 'Active', 'Active', 'Active', 'Inactive',
       'Inactive', 'Inactive', 'Inacti

##  What is the average step count across the dataset?

In [8]:
avg_step_count=np.average(step_count)
avg_step_count

2935.9375

## On which date did the person burn the most calories, and what was their mood on that day?

In [9]:
max_calories_index=np.argmax(calories_burned)
print('index',max_calories_index,data[max_calories_index])


index 69 ('14-12-2017', 7422, 'Happy', 243, 5, 'Active')


## How many days does the dataset show the person being "Active" vs. "Inactive"?

In [10]:
active_days=np.sum(active == 'Active')
print(f'Number of Active days : {active_days}')

Number of Active days : 42


In [11]:
inactive_days=np.sum(active == 'Inactive')
print(f'Number of Inactive days : {inactive_days}')


Number of Inactive days : 54


## Which mood is associated with the highest average calories burned?

In [12]:
avg_calories_sad=np.argwhere(mood=='Sad')

Sad_people=calories_burned[avg_calories_sad] 
print(f'Average Sad person burnes : {np.average(Sad_people)}')


Average Sad person burnes : 67.72413793103448


In [13]:
avg_calories_Neutral=np.argwhere(mood=='Neutral')

Neutral_people=calories_burned[avg_calories_Neutral] 
print(f'Average Neutral person burnes : {np.average(Neutral_people)}')

Average Neutral person burnes : 98.70370370370371


In [14]:
avg_calories_Happy=np.argwhere(mood=='Happy')

Happy_people=calories_burned[avg_calories_Happy] 
print(f'Average Happy person burnes : {np.average(Happy_people)}')

Average Happy person burnes : 108.55


### The Higest avgerage calories burned are Happy people
        Average Happy person burnes : 108.55

        Average Neutral person burnes : 98.70370370370371

        Average Sad person burnes : 67.72413793103448